`tls binding` should handle `null` key returned from `vault`

**Describe the bug**
When `zilla.yaml` is misconfigured or `keystore` is missing a specific `key` entry then the `vault` can return `null` for the `PrivateKeyEntry` causing an exception.

**To Reproduce**
Steps to reproduce the behavior:
1. Checkout `zilla-examples`
2. Go to `tls.echo` example
3. Change `tls` binding `options` `keys` from `localhost` to something else like `localhostx`
4. Run ./setup.sh
5. See error in docker logs

**Expected behavior**
Zilla should report the error.

**Additional context**
Docker log output
```
---
name: example
vaults:
  server:
    type: filesystem
    options:
      keys:
        store: tls/localhost.p12
        type: pkcs12
        password: ${{env.KEYSTORE_PASSWORD}}
bindings:
  tcp_server0:
    type: tcp
    kind: server
    options:
      host: 0.0.0.0
      port: 23456
    exit: tls_server0
  tls_server0:
    type: tls
    kind: server
    vault: server
    options:
      keys:
        - localhostx
      sni:
        - localhost
      alpn:
        - echo
    routes:
      - when:
          - alpn: echo
        exit: echo_server0
  echo_server0:
    type: echo
    kind: server
error
java.util.concurrent.CompletionException: java.lang.NullPointerException: invalid null input
at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:332)
at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:347)
at java.base/java.util.concurrent.CompletableFuture$BiRelay.tryFire(CompletableFuture.java:1498)
at java.base/java.util.concurrent.CompletableFuture$CoCompletion.tryFire(CompletableFuture.java:1219)
at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2162)
at io.aklivity.zilla.runtime.engine@0.9.51/io.aklivity.zilla.runtime.engine.internal.registry.NamespaceTask.run(NamespaceTask.java:53)
at io.aklivity.zilla.runtime.engine@0.9.51/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.onSystemSignal(DispatchAgent.java:905)
at io.aklivity.zilla.runtime.engine@0.9.51/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.onSystemMessage(DispatchAgent.java:855)
at io.aklivity.zilla.runtime.engine@0.9.51/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.handleRead(DispatchAgent.java:1001)
at io.aklivity.zilla.runtime.engine@0.9.51/io.aklivity.zilla.runtime.engine.internal.concurent.ManyToOneRingBuffer.read(ManyToOneRingBuffer.java:181)
at io.aklivity.zilla.runtime.engine@0.9.51/io.aklivity.zilla.runtime.engine.internal.registry.DispatchAgent.doWork(DispatchAgent.java:675)
at org.agrona.core/org.agrona.concurrent.AgentRunner.doDutyCycle(AgentRunner.java:291)
at org.agrona.core/org.agrona.concurrent.AgentRunner.run(AgentRunner.java:164)
at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.NullPointerException: invalid null input
at java.base/java.security.KeyStore.setEntry(KeyStore.java:1580)
at io.aklivity.zilla.runtime.binding.tls@0.9.51/io.aklivity.zilla.runtime.binding.tls.internal.config.TlsBindingConfig.newKeys(TlsBindingConfig.java:394)
at io.aklivity.zilla.runtime.binding.tls@0.9.51/io.aklivity.zilla.runtime.binding.tls.internal.config.TlsBindingConfig.init(TlsBindingConfig.java:91)
at io.aklivity.zilla.runtime.binding.tls@0.9.51/io.aklivity.zilla.runtime.binding.tls.internal.stream.TlsServerFactory.attach(TlsServerFactory.java:216)
at io.aklivity.zilla.runtime.binding.tls@0.9.51/io.aklivity.zilla.runtime.binding.tls.internal.TlsBindingContext.attach(TlsBindingContext.java:57)
at io.aklivity.zilla.runtime.engine@0.9.51/io.aklivity.zilla.runtime.engine.internal.registry.BindingRegistry.attach(BindingRegistry.java:45)
at io.aklivity.zilla.runtime.engine@0.9.51/io.aklivity.zilla.runtime.engine.internal.registry.NamespaceRegistry.attachBinding(NamespaceRegistry.java:141)
at java.base/java.util.Arrays$ArrayList.forEach(Arrays.java:4204)
at io.aklivity.zilla.runtime.engine@0.9.51/io.aklivity.zilla.runtime.engine.internal.registry.NamespaceRegistry.attach(NamespaceRegistry.java:119)
at io.aklivity.zilla.runtime.engine@0.9.51/io.aklivity.zilla.runtime.engine.internal.registry.ConfigurationRegistry.attachNamespace(ConfigurationRegistry.java:159)
at io.aklivity.zilla.runtime.engine@0.9.51/io.aklivity.zilla.runtime.engine.internal.registry.NamespaceTask.run(NamespaceTask.java:48)
... 8 more
```

--- a/runtime/binding-tls/src/main/java/io/aklivity/zilla/runtime/binding/tls/internal/TlsConfiguration.java
+++ b/runtime/binding-tls/src/main/java/io/aklivity/zilla/runtime/binding/tls/internal/TlsConfiguration.java
@@ -15,6 +15,8 @@
  */
 package io.aklivity.zilla.runtime.binding.tls.internal;
 
+import static io.aklivity.zilla.runtime.engine.EngineConfiguration.ENGINE_VERBOSE;
+
 import java.security.KeyStore;
 
 import io.aklivity.zilla.runtime.engine.Configuration;
@@ -30,6 +32,7 @@ public class TlsConfiguration extends Configuration
     public static final BooleanPropertyDef TLS_IGNORE_EMPTY_VAULT_REFS;
     public static final LongPropertyDef TLS_AWAIT_SYNC_CLOSE_MILLIS;
     public static final BooleanPropertyDef TLS_PROACTIVE_CLIENT_REPLY_BEGIN;
+    public static final BooleanPropertyDef TLS_VERBOSE;
 
     private static final ConfigurationDef TLS_CONFIG;
 
@@ -45,6 +48,7 @@ public class TlsConfiguration extends Configuration
         TLS_IGNORE_EMPTY_VAULT_REFS = config.property("ignore.empty.vault.refs", false);
         TLS_AWAIT_SYNC_CLOSE_MILLIS = config.property("await.sync.close.millis", 3000L);
         TLS_PROACTIVE_CLIENT_REPLY_BEGIN = config.property("proactive.client.reply.begin", false);
+        TLS_VERBOSE = config.property("verbose", TlsConfiguration::verboseDefault);
         TLS_CONFIG = config;
     }
 
@@ -99,6 +103,11 @@ public class TlsConfiguration extends Configuration
         return TLS_PROACTIVE_CLIENT_REPLY_BEGIN.get(this);
     }
 
+    public boolean verbose()
+    {
+        return TLS_VERBOSE.getAsBoolean(this);
+    }
+
     private static String cacertsStoreTypeDefault(
         Configuration config)
     {
@@ -109,5 +118,11 @@ public class TlsConfiguration extends Configuration
         Configuration config)
     {
         return System.getProperty("javax.net.ssl.trustStore");
+    }
+
+    private static boolean verboseDefault(
+        Configuration config)
+    {
+        return ENGINE_VERBOSE.getAsBoolean(config);
     }
 }

--- a/runtime/binding-tls/src/main/java/io/aklivity/zilla/runtime/binding/tls/internal/config/TlsBindingConfig.java
+++ b/runtime/binding-tls/src/main/java/io/aklivity/zilla/runtime/binding/tls/internal/config/TlsBindingConfig.java
@@ -20,6 +20,7 @@ import static io.aklivity.zilla.runtime.binding.tls.internal.types.ProxyInfoType
 import static io.aklivity.zilla.runtime.binding.tls.internal.types.ProxyInfoType.AUTHORITY;
 import static io.aklivity.zilla.runtime.binding.tls.internal.types.ProxyInfoType.SECURE;
 import static io.aklivity.zilla.runtime.binding.tls.internal.types.ProxySecureInfoType.NAME;
+import static java.lang.System.currentTimeMillis;
 import static java.util.Collections.singletonList;
 import static java.util.stream.Collectors.toList;
 import static javax.net.ssl.StandardConstants.SNI_HOST_NAME;
@@ -69,7 +70,7 @@ public final class TlsBindingConfig
 
     public final long id;
     public final long vaultId;
-    public final String name;
+    public final String qname;
     public final TlsOptionsConfig options;
     public final KindConfig kind;
     public final List<TlsRouteConfig> routes;
@@ -81,7 +82,7 @@ public final class TlsBindingConfig
     {
         this.id = binding.id;
         this.vaultId = binding.vaultId;
-        this.name = binding.name;
+        this.qname = binding.qname;
         this.kind = binding.kind;
         this.options = binding.options != null ? TlsOptionsConfig.class.cast(binding.options) : OPTIONS_DEFAULT;
         this.routes = binding.routes.stream().map(TlsRouteConfig::new).collect(toList());
@@ -439,6 +440,16 @@ public final class TlsBindingConfig
                 for (String keyName : keyNames)
                 {
                     KeyStore.PrivateKeyEntry entry = vault.key(keyName);
+                    if (entry == null)
+                    {
+                        if (config.verbose())
+                        {
+                            System.out.printf("%d [%s] key \"%s\" not found\n",
+                                    currentTimeMillis(), this.qname, keyName);
+                        }
+                        continue;
+                    }
+
                     KeyStore.ProtectionParameter protection = new KeyStore.PasswordProtection(password);
                     store.setEntry(keyName, entry, protection);
                 }

--- a/runtime/binding-tls/src/test/java/io/aklivity/zilla/runtime/binding/tls/internal/streams/ServerIT.java
+++ b/runtime/binding-tls/src/test/java/io/aklivity/zilla/runtime/binding/tls/internal/streams/ServerIT.java
@@ -94,6 +94,13 @@ public class ServerIT
         k3po.finish();
     }
 
+    @Test
+    @Configuration("server.keys.not.found.yaml")
+    public void shouldIgnoreKeysNotFound() throws Exception
+    {
+    }
+
+
     @Ignore("https://github.com/k3po/k3po/issues/454 - Support connect aborted")
     @Test
     @Configuration("server.yaml")

--- a/specs/binding-tls.spec/src/main/scripts/io/aklivity/zilla/specs/binding/tls/config/server.keys.not.found.yaml
+++ b/specs/binding-tls.spec/src/main/scripts/io/aklivity/zilla/specs/binding/tls/config/server.keys.not.found.yaml
@@ -1,0 +1,35 @@
+#
+# Copyright 2021-2023 Aklivity Inc.
+#
+# Aklivity licenses this file to you under the Apache License,
+# version 2.0 (the "License"); you may not use this file except in compliance
+# with the License. You may obtain a copy of the License at:
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+
+---
+name: test
+vaults:
+    server:
+        type: filesystem
+        options:
+            keys:
+                store: stores/server/keys
+                type: pkcs12
+                password: generated
+bindings:
+    net0:
+        type: tls
+        kind: server
+        vault: server
+        options:
+            keys:
+                - not.found
+        exit: app0

--- a/specs/binding-tls.spec/src/test/java/io/aklivity/zilla/specs/binding/tls/config/SchemaTest.java
+++ b/specs/binding-tls.spec/src/test/java/io/aklivity/zilla/specs/binding/tls/config/SchemaTest.java
@@ -132,6 +132,14 @@ public class SchemaTest
     }
 
     @Test
+    public void shouldValidateServerKeysNotFound()
+    {
+        JsonObject config = schema.validate("server.keys.not.found.yaml");
+
+        assertThat(config, not(nullValue()));
+    }
+
+    @Test
     public void shouldValidateServerMutual()
     {
         JsonObject config = schema.validate("server.mutual.yaml");

